src/renderers/canvas.js

```diff
@@ -41,6 +41,38 @@ CanvasRenderer.prototype.circleStroke = function(left, top, size, color, stroke,
     this.ctx.stroke();
 };
 
+CanvasRenderer.prototype.shadow = function(shape, shadows) {
+    var parseShadow = function(str) {
+        var propertyFilters = { color: /^(#|rgb|hsl|(?!(inset|initial|inherit))\D+)/i, inset: /^inset/i, px: /px$/i };
+        var pxPropertyNames = [ 'x', 'y', 'blur', 'spread' ];
+        var properties = str.split(/ (?![^(]*\))/);
+        var info = {};
+        for (var key in propertyFilters) {
+            info[key] = properties.filter(propertyFilters[key].test.bind(propertyFilters[key]));
+            info[key] = info[key].length === 0 ? null : info[key].length === 1 ? info[key][0] : info[key];
+        }
+        for (var i=0; i<info.px.length; i++) {
+            info[pxPropertyNames[i]] = parseInt(info.px[i]);
+        }
+        return info;
+    };
+    var drawShadow = function(shadow) {
+        var info = parseShadow(shadow);
+        if (!info.inset) {
+            context.shadowOffsetX = info.x;
```

Comment:
```suggestion
            if (info.spread > 0) {
                for (var i = -1; i < 2; i += 2) {
                    for (var i2 = 1; i2 > -2; i2 -= 2) {
                        context.shadowOffsetX = info.x + i * info.spread;
                        context.shadowOffsetY = info.y + i2 * info.spread;
                        context.shadowColor = info.color;
                        context.shadowBlur = info.blur;
                        context.fill();
                    }
                }
                return;
            }
            context.shadowOffsetX = info.x;
```
When the spread is greater 0 you can draw the shape 4 times to simulate the spread. As long as the spread value is not bigger than double of the actual width / height it should work as intended (or better than not handling it at all).
But it is not possible to do negative spread with this method,